Assets/Character/Scripts/PlayerStateController.cs

```diff
@@ -83,10 +86,9 @@ bool PressedMoveRight()
 
 	void IdleStateUpdate()
 	{
-		if (!IsGrounded())
-		{
+		// If a default case fell here, don't respond to input until done with animation
+		if (!PlayingNextAnimation())
 			return;
-		}
 
 		if (PressedRoll())

```

Comment:
Create another file for the input map. Another file that defines Space as roll, A as mid attack, B as  pressed Jump etc. Something that sam would know how to change